src/test/java/BaseTest.java

```diff
@@ -1,11 +1,138 @@
+        avatar.click();
+    }
+    public void provideCurrentPassword () {
+        WebElement currentPassword = driver.findElement(By.cssSelector("input[type='password']"));
+        currentPassword.click();
+        currentPassword.clear();
+        currentPassword.sendKeys("te$t$tudent");
+    }
+    public String generateRandomName () {
+        return RandomStringUtils.randomAlphabetic(5,10);
+    }
+    public void provideProfileName (String randomName) {
+        WebElement profileName = driver.findElement(By.cssSelector("input[name='name']"));
+        profileName.click();
+        profileName.clear();
+        profileName.sendKeys(randomName);
+    }
+    public void clickSaveProfile () {
+        WebElement saveProfile = driver.findElement(By.cssSelector("button[class='btn-submit']"));
+        saveProfile.click();
+    }
+
+    //HELPER METHODS for Search Songs
+    public void searchSong(String song) throws InterruptedException {
+    WebElement searchBar = driver.findElement(By.cssSelector("input[type='search']"));
+    searchBar.click();
+    searchBar.clear();
+    searchBar.sendKeys(song);
+    Thread.sleep(2000);
+    }
+    public void clickViewAllSearchedSongs() throws InterruptedException {
+        WebElement viewAllSearchedSongs = driver.findElement(By.cssSelector("button[data-test='view-all-songs-btn']"));
+        viewAllSearchedSongs.click();
+        Thread.sleep(2000);
+    }
+    public void pickFirstSong() throws InterruptedException {
+        WebElement firstSong = driver.findElement(By.xpath("//section[@id='songResultsWrapper']//tr[@class='song-item'][1]"));
+        firstSong.click();
+        Thread.sleep(2000);
+    }
+    public void clickAddToButton() throws InterruptedException{
+        WebElement addToButton = driver.findElement(By.cssSelector("button[class='btn-add-to']"));
+        addToButton.click();
+        Thread.sleep(2000);
+    }
+        public void chooseUniquePlaylist() throws InterruptedException{
+        WebElement uniquePlaylist = driver.findElement(By.xpath("//section[@id='songResultsWrapper']//li[contains(text(),'Unique Playlist')]"));
+        uniquePlaylist.click();
+        Thread.sleep(1000);
+    }
+
+    public String getActualSongAddedMessage() {
+        WebElement actualSongAddedMessage = driver.findElement(By.cssSelector("div[class='success show']"));
+        return actualSongAddedMessage.getText();
+    }
+
+    // HELPER METHOD for playing songs
+    public void clickPlay() {
+        WebElement playButton = driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+        clickPlayNext();
+        playButton.click();
+    }
+    public void clickPlayNext() {
+        WebElement playNextButton = driver.findElement(By.cssSelector("[data-testid='play-next-btn']"));
+        playNextButton.click();
+    }
+    public boolean isSongPlaying() {
+        WebElement songBars = driver.findElement(By.cssSelector("[alt='Sound bars']"));
+        return songBars.isDisplayed();
```

Comment:
The test executes, but the assertion fails whether the song is playing or not. for that, we have to wait for a certain time to get the sound bar visible after clicking on the play button. Using Thread.sleep will overcome this issue.
Though Threas.sleep is not a good practice to be used. I believe you will be learning about waiting in future classes.
```
public boolean isSongPlaying() throws InterruptedException {
        Thread.sleep(3000);
        WebElement songBars = driver.findElement(By.cssSelector("[alt='Sound bars']"));
        return songBars.isDisplayed();
    }
```